File: easy/0747_largest_number_at_least_twice_of_others/solution.cpp
#include <algorithm>
#include <ranges>
#include <vector>

class Solution {
public:
  int dominantIndex(std::vector<int> &nums) {
    auto it = std::ranges::max_element(nums);

    for (const auto &num : nums) {
      if (num * 2 > *it && num != *it)
        return -1;
    }

    return static_cast<int>(std::distance(nums.begin(), it));
  }
};

File: easy/0747_largest_number_at_least_twice_of_others/solution.py
class Solution:
    def dominantIndex(self, nums: list[int]) -> int:
        maxIdx = 0
        maxNum = nums[0]

        for i, num in enumerate(nums):
            if num > maxNum:
                maxIdx = i
                maxNum = num

        if any(num * 2 > maxNum for num in nums if num != maxNum):
            return -1

        return maxIdx
